### ISSUE TITLE
no-unused-vars wrong error when using imported JSDoc definition as @property type.

### ISSUE THREAD

**Issue by ernestostifano** [eslint/eslint#13075]
**Tell us about your environment**

* ***Node version:*** v13.11.0
* ***npm version:*** v6.13.7
* ***Local ESLint version:*** v6.8.0 (Currently used)
* ***Global ESLint version:*** Not found

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel-ESLint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
const browserPackages = [
    /* ...somePaths */
];

const nodePackages = [
    /* ...somePaths */
    /* ISSUE FILE INCLUDED HERE */
];

const reactFlowPackages = [ /* ...somePaths */ ];

const currentNodeVersion = '>=13.11.0';

module.exports = {
    root: true,
    noInlineConfig: true,
    parser: 'babel-eslint',
    parserOptions: {
        ecmaVersion: 2020,
        sourceType: 'module'
    },
    plugins: ['prettier', 'jsdoc', 'import'],
    extends: [
        'eslint:recommended',
        'plugin:prettier/recommended',
        'plugin:jsdoc/recommended',
        'plugin:import/errors',
        'plugin:import/warnings'
    ],
    env: {
        es2017: true,
        mocha: true,
        'shared-node-browser': true
    },
    rules: {
        'no-unused-vars': ['error', {ignoreRestSiblings: true}],
        'padding-line-between-statements': [
            'off',
            {blankLine: 'always', prev: 'block-like', next: '*'},
            {blankLine: 'always', prev: '*', next: 'block-like'}
        ],
        'no-prototype-builtins': 'off',
        'prettier/prettier': 'warn',
        'jsdoc/valid-types': 'error',
        'jsdoc/check-indentation': 'warn',
        'jsdoc/check-tag-names': 'error',
        'jsdoc/check-param-names': 'error',
        'jsdoc/check-property-names': 'error',
        'jsdoc/require-jsdoc': [
            'warn',
            {
                require: {
                    ArrowFunctionExpression: true,
                    ClassDeclaration: true,
                    ClassExpression: true,
                    FunctionDeclaration: true,
                    FunctionExpression: true,
                    MethodDefinition: true
                },
                exemptEmptyFunctions: true
            }
        ],
        'jsdoc/require-description': [
            'warn',
            {
                descriptionStyle: 'tag',
                exemptedBy: ['ignore', 'see', 'classdesc', 'constructs', 'type']
            }
        ],
        'jsdoc/require-description-complete-sentence': [
            'warn',
            {
                tags: ['description']
            }
        ],
        'jsdoc/require-hyphen-before-param-description': ['warn', 'always'],
        'jsdoc/require-param': [
            'warn',
            {
                exemptedBy: ['ignore', 'see', 'classdesc', 'constructs', 'type']
            }
        ],
        'jsdoc/require-param-name': 'error',
        'jsdoc/require-param-type': 'error',
        'jsdoc/require-param-description': 'warn',
        'jsdoc/require-property': 'warn',
        'jsdoc/require-property-name': 'warn',
        'jsdoc/require-property-type': 'warn',
        'jsdoc/require-property-description': 'warn',
        'jsdoc/require-returns': [
            'warn',
            {
                exemptedBy: ['ignore', 'see', 'classdesc', 'constructs', 'type']
            }
        ],
        'jsdoc/require-returns-type': 'error',
        'jsdoc/require-returns-description': 'warn',
        'import/no-unresolved': 'warn',
        'import/named': 'warn',
        'import/no-absolute-path': 'warn',
        'import/no-self-import': 'warn',
        'import/no-cycle': 'warn',
        'import/no-useless-path-segments': 'warn',
        'import/export': 'warn',
        'import/no-deprecated': 'warn',
        'import/no-mutable-exports': 'warn',
        'import/extensions': ['warn', 'ignorePackages'],
        'import/no-default-export': 'warn',
        'import/group-exports': 'warn',
        'import/order': [
            'warn',
            {
                groups: [
                    'builtin',
                    ['external', 'unknown'],
                    'internal',
                    'parent',
                    'sibling',
                    'index'
                ],
                alphabetize: {
                    order: 'asc'
                }
            }
        ],
        'import/newline-after-import': 'warn',
        'import/no-namespace': 'off',
        'import/no-duplicates': 'warn',
        'import/exports-last': 'warn'
    },
    settings: {
        jsdoc: {
            mode: 'jsdoc',
            tagNamePreference: {
                class: {
                    message:
                        'Class tag may provoke duplicated documentation. Use only @classdesc to document classes.'
                },
                returns: 'returns',
                property: 'property',
                param: 'param',
                augments: 'extends',
                description: 'description',
                function: 'function'
            },
            overrideReplacesDocs: false,
            preferredTypes: {
                object: 'Object'
            },
            ignorePrivate: true
        }
    },
    overrides: [
        // FILES OUTSIDE SRC
        {
            files: ['!**/src/**/*.js'],
            excludedFiles: [
                './packages/*/stories/**/*.js',
                './packages/*/.storybook/**/*.js'
            ],
            plugins: ['node'],
            extends: ['plugin:node/recommended-script'],
            parserOptions: {
                ecmaVersion: 2015,
                sourceType: 'script'
            },
            env: {
                es2017: false,
                es6: true,
                node: true
            }
        },
        // COMMON JS
        {
            files: [
                './packages/*/stories/**/*.js',
                './packages/*/.storybook/**/*.js'
            ],
            env: {
                'shared-node-browser': false,
                browser: true,
                node: true,
                commonjs: true
            }
        },
        // REACT + FLOW PACKAGES
        {
            files: reactFlowPackages,
            parserOptions: {
                ecmaFeatures: {
                    jsx: true
                }
            },
            plugins: ['flowtype', 'react'],
            extends: [
                'prettier/flowtype',
                'prettier/react',
                'plugin:flowtype/recommended',
                'plugin:react/recommended'
            ],
            rules: {
                'jsdoc/check-tag-names': [
                    'warn',
                    {
                        definedTags: ['flow']
                    }
                ],
                'react/require-default-props': 'error'
            },
            settings: {
                flowtype: {
                    onlyFilesWithFlowAnnotation: true
                },
                react: {
                    version: 'detect'
                }
            }
        },
        // BROWSER PACKAGES
        {
            files: browserPackages,
            env: {
                'shared-node-browser': false,
                browser: true
            }
        },
        // NODE PACKAGES
        {
            files: nodePackages,
            plugins: ['node'],
            extends: ['plugin:node/recommended-module'],
            globals: {
                __dirname: 'readonly',
                __filename: 'readonly',
                module: 'readonly'
            },
            env: {
                'shared-node-browser': false,
                node: true
            },
            rules: {
                'node/shebang': [
                    'error',
                    {
                        convertPath: {
                            'src/bin/**/*.js': ['^src/(.+?)\\.js$', 'lib/$1.js']
                        }
                    }
                ],
                'node/no-unsupported-features/es-syntax': [
                    'error',
                    {
                        version: currentNodeVersion,
                        ignores: ['dynamicImport', 'modules']
                    }
                ],
                'node/no-unsupported-features/node-builtins': [
                    'error',
                    {
                        version: currentNodeVersion
                    }
                ]
            }
        }
    ]
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

**REPLICA:**
```js
import {myTypesA} from '../internal/file.js'; // ERROR
import {myTypesB} from '../internal/file.js'; // NO ERROR

/*
* @typedef newType
* @property {myTypesA.someType} someProp - Some prop.
*/

/*
* @param {newType} arg - Arg.
*/
function myFunctionA(arg) {
    return arg;
}

/*
* @param {myTypesB.someType} arg - Arg.
*/
function myFunctionB(arg) {
    return arg;
}

```

```bash
eslint \"./packages/*/src/**/*.js\" --cache
```

**What did you expect to happen?**

There should not be an error.
JSDoc definition is correctly imported and used even when the error is present.
The error only appears when imported type is used as a type for a `@property` tag.
Error does not appear is used in `@param` rag.

**What actually happened? Please include the actual, raw output from ESLint.**
```js
/path/to/issue/file.js
  1:9  error  'fiberRegExpTypes' is defined but never used  no-unused-vars

```

**Are you willing to submit a pull request to fix this bug?**

I'm currently not familiar with ESLint codebase, so I can't help right now.

**Comment by kaicataldo** [comment]
Thanks for the report. This is working as intended, because ESLint core rules have no knowledge of JSDoc. You'll want to use the [`eslint-plugin-jsdoc/no-undefined-types`](https://www.npmjs.com/package/eslint-plugin-jsdoc#no-undefined-types) to fix your issue.
